pkgs/by-name/li/libsecret/package.nix

```diff
@@ -99,25 +115,80 @@ stdenv.mkDerivation rec {
     patchShebangs ./tool/test-*.sh
   '';
 
-  preCheck = ''
-    # Our gobject-introspection patches make the shared library paths absolute
-    # in the GIR files. When running tests, the library is not yet installed,
-    # though, so we need to replace the absolute path with a local one during build.
-    # We are using a symlink that will be overwitten during installation.
-    mkdir -p $out/lib $out/lib
-    ln -s "$PWD/libsecret/libmock-service.so" "$out/lib/libmock-service.so"
-    ln -s "$PWD/libsecret/libsecret-1.so.0" "$out/lib/libsecret-1.so.0"
+  preConfigure = lib.optionalString abrmdSupport ''
+    # Add dependencies on TCTI modules required for user‐space TPM resource
+    # manager support so that they can be loaded at run‐time.
+    mesonFlagsArray+=("-Dc_link_args=-Wl,--push-state,--no-as-needed -ltss2-tcti-tabrmd -ltss2-tcti-device -Wl,--pop-state")
   '';
 
-  checkPhase = ''
-    runHook preCheck
-
-    dbus-run-session \
-      --config-file=${dbus}/share/dbus-1/session.conf \
-      meson test --print-errorlogs --timeout-multiplier 0
-
-    runHook postCheck
-  '';
+  preCheck =
+    ''
+      # Our gobject-introspection patches make the shared library paths absolute
+      # in the GIR files. When running tests, the library is not yet installed,
+      # though, so we need to replace the absolute path with a local one during build.
+      # We are using a symlink that will be overwitten during installation.
+      mkdir -p $out/lib $out/lib
+      ln -s "$PWD/libsecret/libmock-service.so" "$out/lib/libmock-service.so"
+      ln -s "$PWD/libsecret/libsecret-1.so.0" "$out/lib/libsecret-1.so.0"
+    ''
+    + lib.optionalString withTpm2Tss ''
+      export XDG_CONFIG_HOME="$(mktemp -d)"
+
+      export swtpm="$(mktemp -d)"
+      mkdir "$swtpm/state"
+
+      swtpm_setup --create-config-files --root
+      swtpm_setup \
+        --tpm-state "$swtpm/state" \
+        --tpm2 \
+        --createek \
+        --allow-signing \
+        --decryption \
+        --create-ek-cert \
+        --create-platform-cert \
+        --lock-nvram
+    '';
+
+  checkPhase =
+    let
+      checkScript =
+        lib.optionalString withTpm2Tss ''
+          swtpm socket \
+            --server "type=unixio,path=$swtpm/sock" \
+            --daemon \
+            --ctrl "type=unixio,path=$swtpm/sock.ctrl" \
+            --tpmstate "dir=$swtpm/state" \
+            --tpm2 \
+            --flags startup-clear
+
+          trap "swtpm_ioctl --unix '$swtpm/sock.ctrl' -s" EXIT
+
+          tpm2-abrmd \
+            --tcti="swtpm:path=$swtpm/sock" \
+            --allow-root \
+            --flush-all \
+            --session &
+
+          export TCTI="tabrmd:bus_type=session"
+        ''
+        + lib.optionalString (!abrmdSupport) ''
```

Comment:
Why is this negated?

Reply:
It is only needed if the `tabrmd` TCTI module has not already been linked during compile time.

`abrmdSupport`:
  - `true`: Test binaries will locate the `tabrmd` module through the `DT_RUNPATH` and `DT_NEEDED` entries in `libsecret-1.so` added at compile‐time. Allow tests to fail if not properly linked.
  - `false`: Test binaries will locate the `tabrmd` module through the `LD_LIBRARY_PATH` environment variable. It is only required during testing because the TPM emulator `swtpm` lacks an integrated resource manager.

Reply:
Could you please clarify that in the comment? It looks confusing that the abrmd package is used when the corresponding argument is false.